modules/payments/modules/payments/paymentHistoryModal.js

```diff
@@ -0,0 +1,130 @@
+import React, { PureComponent, useState, useEffect } from 'react';
+import { View, Text, Modal, StyleSheet, Pressable, Alert, RefreshControl } from 'react-native';
+import { FlatList, TouchableOpacity } from 'react-native-gesture-handler';
+import { fetchPaymentHistory } from "./api";
+import { Dimensions } from 'react-native'
+let deviceWidth = Dimensions.get('window').width
+
+
+const PaymentHistoryModal = (props) => {
+
+    const [paymentHistory, setPaymentHistory] = useState([]);
+    const [modalVisible, setModalVisible] = useState(false);
+    const [refresh, setRefresh] = useState(false);
+
+    useEffect(() => {
+        get_payments();
+    }, [])
+
+    const get_payments = async () => {
+        setRefresh(true);
+        const res = await fetchPaymentHistory()
+        fetchPaymentHistory().then(res => {
+            setRefresh(false);
+            setPaymentHistory(res);
+        })
+    }
```

Comment:
If you return a **Promise** in the `fetchPaymentHistory` function, you could simplify this to:

```suggestion
    useEffect(() => {
      fetchPaymentHistory(userToken)
        .then(response => response.json())
        .then(json => {
          const { status, data } = json;
          setRefresh(false);
          setPaymentHistory({ status, data });
        });
    }, [])
```